packages/cli/src/services/check-runner.ts

```diff
@@ -201,6 +196,20 @@ export default class CheckRunner extends EventEmitter {
     })
   }
 
+  private setAllTimeouts () {
+    Array.from(this.checks.entries()).forEach(([checkRunId, check]) =>
+      this.timeouts.set(checkRunId, setTimeout(() => {
+        this.timeouts.delete(checkRunId)
```

Comment:
@umutuzgur , I would remove this line and always keep the timeout instance. Then, in the  `disableTimeout()`, I would add an `if` condition like:
```ts
if (timeout) {
  clearTimeout(timeout)
}
```